packages/react-admin-core/src/stack/Switch.tsx

```diff
@@ -35,44 +35,81 @@ function useUuid() {
         ref.current = UUID.v4() as string;
     }
     return ref.current;
-} 
+}
+
+export function useStackSwitch() {
+    const apiRef = React.useRef<IStackSwitchApi>();
+    const id = useUuid();
+    const api: IStackSwitchApi = {
+        id,
+        activatePage: (pageName: string, payload: string, subUrl?: string) => {
+            apiRef.current?.activatePage(pageName, payload, subUrl);
+        },
+        updatePageBreadcrumbTitle: (title?: string) => {
+            apiRef.current?.updatePageBreadcrumbTitle(title);
+        }
+    }
+    const StackSwitchWithHookProps = React.useMemo(() => {
+        return (props: IProps) => {
+            return <StackSwitch {...props} id={id} injectApi={(a: IStackSwitchApi) => {
```

Comment:
You could use the ref-pattern from #205. 
```ts
    const apiRef = React.useRef<IStackSwitchApi>();

            return <StackSwitch {...props} id={id} ref={apiRef}

```

So in fact we could leave the version #205 as it is an add the useStackSwitch as a convenient-api, so nobody needs to use the verbose useImperativeHandle in userland, instead you can use the hook.
In terms of ugliness both versions (useImperativeHandle, injectApi) are equal.

Reply:
much easier, changed that. And i also added a (now fully compatible!!) StackSwitch export.